scripts/wallet.js

```diff
@@ -1040,11 +1040,20 @@ export class Wallet {
         // Add primary output
         if (isDelegation) {
             if (!returnAddress) [returnAddress] = this.getNewAddress(1);
-            transactionBuilder.addColdStakeOutput({
-                address: returnAddress,
-                addressColdStake: address,
-                value,
-            });
+            // The per-output target for maximum staking efficiency
+            const nTarget = cChainParams.current.stakeSplitTarget;
+            // Generate optimal staking outputs
+            for (let i = 0; i <= Math.floor(value / nTarget); i++) {
```

Comment:
I see the problem here, but I would prefer that the whole `value` is staked... what about something like this?

```suggestion
            if(value < COIN) {
                throw new Error("below consensus);
            } else if (value < nTarget) {
                transactionBuilder.addColdStakeOutput({
                    address: returnAddress,
                    addressColdStake: address,
                    value,
                });
            } else {
                for (let i = 0; i < Math.floor(value / nTarget); i++) {
                    transactionBuilder.addColdStakeOutput({
                        address: returnAddress,
                        addressColdStake: address,
                        value: i == 0 ? nTarget + (value % nTarget) : nTarget,
                });
                }
            }
```